### PR TITLE
[ci][wins3/] support volumes for windows container

### PR DESCRIPTION
**Pull request by can-anyscale**
Context: this is a series of PR to move windows wheel to rayci (and deprecate .buildkite/windows_ci.sh). The mentioned script runs into issues when I try to upgrade it to python 3.11. Moving the script to rayci to uniform the window build environment.

Support volumes when creating a windows container. I do this by moving `volumes` to the base container class, so that both linux and windows container and initialize volumes. `volumes` is required in a later PR when I use the windows container to build windows wheel.

Test:
- CI

### CODE REVIEW

**Review by aslonnie — APPROVED** [review]

**Comment by aslonnie** [comment]
(ready to merge? I can review the next PR on top of this now?)

**Comment by can-anyscale** [comment]
@aslonnie you can review https://github.com/ray-project/ray/pull/43465 yess, i'll merge them together after approval (to save some rebase + rerun of CI))

**Pull request merged**
